keyring/system.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package keyring
+
+import (
+	"encoding/base64"
+
+	"github.com/zalando/go-keyring"
+)
+
+var _ Keyring = (*systemKeyring)(nil)
+
+// systemKeyring is a keyring that utilizies the
+// built in key management system of the OS.
+type systemKeyring struct{}
+
+func openSystemKeyring() *systemKeyring {
+	return &systemKeyring{}
+}
+
+func (systemKeyring) Set(name string, key []byte) error {
+	enc := base64.StdEncoding.EncodeToString(key)
+	return keyring.Set(service, name, enc)
```

Comment:
todo: Adding comment here, but it applies to all the system Get/Set calls.

By using a static service name here, then if you have two nodes running on the same machine, that both want to use a the system store, their keys will overlap.

Can either
A) Use a paramaterized service name via the config
B) Use the single service name as you have here, but add a namespace on the item names during the `Get/Set`, which is set via the config.
C) Anything else you think of ;)

Reply:
Decided to go with option A.